tests/unit/remote/test_http.py

```diff
@@ -125,3 +125,25 @@ def test_http_method(dvc):
     assert tree._auth_method() == auth
     assert tree.method == "PUT"
     assert isinstance(tree._auth_method(), HTTPBasicAuth)
+
+
+def test_exists(dvc, http, mocker):
+    import io
+
+    import requests
+
+    res = requests.Response()
+    res.raw = io.StringIO("foo")
+    tree = HTTPTree(dvc, http.config)
+
+    mocker.patch.object(tree, "request", return_value=res)
+
+    res.status_code = 200
+    assert tree.exists(http / "file.txt") is True
```

Comment:
Also, since we don't have `http` now, we need to create `url` that's a `URLInfo` (it comes from `dvc.path_info`).

```python
    url = URLInfo("https://example.org/file.txt")
```

And, now, pass those url to the `exists`:
```suggestion
    assert tree.exists(url) is True
```

Reply:
I've added a commit with this change